add a repository layer error handling (#145)
genesis-tech-tribe/nishiki-backend: backend/main/README.md

```shell
$ npm run typedoc:watch
```

## Structure of Application

This application is influenced by Clean Architecture and Domain Driven Development.
Therefore, sometimes that jargon is used inside the code.

Fundamentally, this application is divided into; Controller, Repository, Domain, and UseCase.
The main logic is encapsulated in the Domain and UseCase.
Other parts are used to interact with outside of this application.

Inside this application, the application is divided by domain.
Each domain has the above functions.(Controller, Repository and, so on...)

### HTTP Request Handling from API Gateway

This application is connected to the API Gateway.
Thus, this application needs to act as the REST API.
Those behavior are defined in the `Router`s.
Once the Router gets a request from a client, the router calls the appropriate controller.

![router-structure](./img/routers.drawio.svg)

### Error Handling

Most of the errors that occur in this application are handled explicitly using [result-type](https://github.com/hitohata/result-ts-type) without try-catch syntax.

> But trying to handle every error the same way, cause the increasing cost of implementing without few benefit.

Reviewer: I think it's better to say `increase in noun` instead of `increasing noun`, or just use `increase` as a verb.
```suggestion
But trying to handle every error the same way causes an increase in the cost of implementation without few benefits.
```

Therefore, this application admits to throwing errors inside the external communication process.
Because those errors are out of control in this application.

> That would be the repository and controller layer only.

Reviewer: This `that` refers to the admission, so it would be better to change the verb.
```suggestion
This applies to the repository and controller layers only.
```

The repository layer is called inside the use case that is called by the controller.

> To catch this error, in the controller wraps calling logic by the try-catch.

Reviewer: I'm unsure the subject should be the controller, but I want to clear a subject in the sentence.
```suggestion
To catch their error, the controller wraps calling logic in the try-catch.
```
